springboard_iogt/views.py

```diff
@@ -0,0 +1,70 @@
+import random
+from datetime import datetime
+from itertools import chain
+
+from pyramid.view import view_config
+from elasticutils import F
+
+from springboard.views.base import SpringboardViews
+
+from springboard_iogt.utils import randomize_query
+
+
+class IoGTViews(SpringboardViews):
+
+    @view_config(route_name='home',
+                 renderer='springboard_iogt:templates/home.jinja2')
+    def index_view(self):
+        return self.context(recent_content=self.recent_content())
+
+    def recent_content(self):
+        # get 2 most recent pages and their categories
+        [page1, page2] = self.all_pages.filter(
+            language=self.language).order_by('-created_at')[:2]
+        categories = self.all_categories.filter(
+            uuid__in=filter(lambda uuid: uuid is not None,
+                            [page1.primary_category, page2.primary_category]))
+        categories = dict((category.uuid, category) for category in categories)
+        category1 = (categories[page1.primary_category]
+                     if page1.primary_category else None)
+        category2 = (categories[page2.primary_category]
+                     if page2.primary_category else None)
+
+        # random seed that changes hourly
+        seed = datetime.utcnow().replace(
+            minute=0, second=0, microsecond=0)
+        seed = (seed - datetime.utcfromtimestamp(0)).total_seconds()
+        seed = int(seed)
```

Comment:
Actually this can simply be `seed.hour`. So we'd cycle through 24 unique seeds a day. I've been generating a unique seed per hour *forever*, which is overkill.